byzcoin/service.go

```diff
@@ -2172,17 +2178,22 @@ func (s *Service) addError(tx ClientTransaction, err error) {
 // also returns the temporary StateTrie with the StateChanges applied. Any data
 // from the trie should be read from sst and not the service.
 func (s *Service) processOneTx(sst *stagingStateTrie, tx ClientTransaction,
-	scID skipchain.SkipBlockID) (StateChanges, *stagingStateTrie, error) {
+	scID skipchain.SkipBlockID, timestamp int64) (StateChanges, *stagingStateTrie, error) {
 
 	// Make a new trie for each instruction. If the instruction is
 	// sucessfully implemented and changes applied, then keep it
 	// otherwise dump it.
 	sst = sst.Clone()
+
+	// convert ReadOnlyStateTrie to a GlobalState so that contracts may cast it if they wish
+	roSC := newROSkipChain(s.skService(), scID)
```

Comment:
did you move it there from `executeInstruction` for performance sake?

Reply:
Not really, just because it looked like it could be done earlier, and also it would remove the need to pass the `scID` parameter (which I forgot to do in the commit :grimacing:, coming up).